veps/sig-compute/target-xml-mutation-hooks.md

```diff
@@ -0,0 +1,120 @@
+# VEP 141: Target-Side Pre-Migration Hooks for Domain XML Modification
+
+## Release Signoff Checklist
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+KubeVirt currently applies domain XML transformations on the source launcher pod during live migration.  
+This proposal introduces a target-side pre-migration hook system that performs all mutations on the target launcher pod before the migration start.
+
+## Motivation
+
+Currently, KubeVirt modifies the domain XML on the source during live migration via `migratableDomXML()`. This approach has limitations:
+
+1. **Upgrade**  
+   - New XML transformation logic requires two migrations: old source → new target, then new source → new target.  
+   - Users might not even know a second migration is required, and no automation exists to trigger it during upgrades.
+
+2. **Information Gathering Constraints**  
```

Comment:
Please consider the following framing:
By shifting the domain mutation to the target virt-launcher, we can simplify the overall migration workflow and deprecate some API fields in the VMI status API.